source/Halibut/Transport/Streams/NetworkTimeoutStream.cs

```diff
@@ -200,5 +200,24 @@ public override long Position
             get => inner.Position;
             set => inner.Position = value;
         }
+
+        public async Task<bool> DataAvailable(CancellationToken cancellationToken)
```

Comment:
I think this can be a pass-through if supported property rather than being wrapped in the timeout and cancellation logic. Will change and ensure it doesn't regress the timeout tests

Reply:
Where you finding that `DataAvailable` was blocking?

another trick people use is ask to read into an empty array, although I suspect our own streams will struggle to support that.

Reply:
> Where you finding that DataAvailable was blocking

I don't think it was the cause that turned out to be another bug. This was leftover from trying to fix the issue